### PR TITLE
Adding codec struct tags to SuggestedParams struct

### PR DESCRIPTION
**Pull request by barnjamin**
This struct does not go over the wire so codec struct tags are not required but adding them here to make the json output match documentation

### CODE REVIEW

**Comment by fabrice102** [comment]
That is with regards to https://developer.algorand.org/docs/build-apps/connect/#check-suggested-transaction-parameters
Currently the JSON output does not match the documentation above.

File: types/transaction.go [hunk 1]
```diff
@@ -164,31 +164,30 @@ type SuggestedParams struct {
...
+	LastRoundValid Round `codec:"last-round"`
 
 	// ConsensusVersion indicates the consensus protocol version
 	// as of LastRound.
-	ConsensusVersion string
+	ConsensusVersion string `codec:"consensus-version"`
 
-	// FlatFee indicates whether the passed fee is per-byte or per-transaction
 	// If true, txn fee may fall below the MinTxnFee for the current network protocol.
 	FlatFee bool
```
Comment (jasonpaulos) [inline types/transaction.go:186]:
```suggestion
	FlatFee bool `codec"flat-fee"`
```

File: types/transaction.go [hunk 2]
```diff
@@ -164,31 +164,30 @@ type SuggestedParams struct {
...
 
 	// ConsensusVersion indicates the consensus protocol version
 	// as of LastRound.
-	ConsensusVersion string
+	ConsensusVersion string `codec:"consensus-version"`
 
-	// FlatFee indicates whether the passed fee is per-byte or per-transaction
 	// If true, txn fee may fall below the MinTxnFee for the current network protocol.
-	FlatFee bool
+	FlatFee bool `codec"flat-fee"`
```
Comment (barnjamin) [inline types/transaction.go:186]: This is missing the  colon between tag and value

Comment (jasonpaulos) [reply types/transaction.go]: Thanks for catching that!

File: types/transaction.go [hunk 3]
```diff
@@ -164,31 +164,30 @@ type SuggestedParams struct {
...
+	FirstRoundValid Round `codec:"first-round"`
 
 	// LastRoundValid is the final protocol round on which the txn may be committed
-	LastRoundValid Round
+	LastRoundValid Round `codec:"last-round"`
 
 	// ConsensusVersion indicates the consensus protocol version
 	// as of LastRound.
-	ConsensusVersion string
+	ConsensusVersion string `codec:"consensus-version"`
```
Comment (jasonpaulos) [inline types/transaction.go:184]:
```suggestion

// FlatFee indicates whether the passed fee is per-byte or per-transaction
```

File: types/transaction.go [hunk 4]
```diff
@@ -164,31 +164,31 @@ type SuggestedParams struct {
...
 
-	// FlatFee indicates whether the passed fee is per-byte or per-transaction
+// FlatFee indicates whether the passed fee is per-byte or per-transaction
```
Comment (jasonpaulos) [inline types/transaction.go:185]:
```suggestion
	// FlatFee indicates whether the passed fee is per-byte or per-transaction
```

**Review by jasonpaulos — APPROVED** [review]
Thank you for this contribution! Sorry for the delay in review. This will be included in the next release.

**Pull request merged**